integration_test/tests/power_bi_tests.rs

```diff
@@ -244,4 +246,149 @@ mod integration {
+    /// data retreival flow is:
+    ///     - SQLAllocHandle(SQL_HANDLE_STMT)
+    ///     - SQLGetInfoW(SQL_DRIVER_ODBC_VER)
+    ///     - SQLGetInfoW(SQL_DRIVER_NAME)
+    ///     - SQLExecDirectW({NullTerminatedQuery},SQL_NTS)
+    ///     - SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+    ///     - SQLGetInfoW(SQL_GETDATA_EXTENSIONS)
+    ///     - SQLNumResultCols()
+    ///     - For columns 1 to {numCols}
+    ///         - SQLColAttributeW(SQL_DESC_CONCISE_TYPE)
+    ///         - SQLColAttributeW(SQL_DESC_UNSIGNED)
+    ///         - SQLColAttributeW(SQL_COLUMN_NAME)
+    ///         - SQLColAttributeW(SQL_COLUMN_NULLABLE)
+    ///         - SQLColAttributeW(SQL_DESC_TYPE_NAME)
+    ///         - SQLColAttributeW(SQL_COLUMN_LENGTH)
+    ///         - SQLColAttributeW(SQL_COLUMN_SCALE)
+    ///     - For X rows or until SQLFetch returns SQL_NO_DATA
+    ///         - SQLFetch()
+    ///         - For columns 1 to {numCols}
+    ///             - SQLGetData({colIndex}, {defaultCtoSqlType})
+    ///     - SQLMoreResults()
+    ///     - SQLFreeHandle(SQL_HANDLE_STMT)
+    #[test]
+    fn test_data_retrieval() {
+        let env_handle: HEnv = setup();
+        let (conn_handle, _, _, _) = power_bi_connect(env_handle);
+        let mut stmt: Handle = null_mut();
+        unsafe {
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLAllocHandle(
+                    HandleType::Stmt,
+                    conn_handle as *mut _,
+                    &mut stmt as *mut Handle
+                )
+            );
+
+            // SQL_DRIVER_ODBC_VER and SQL_DRIVER_NAME are not available through odbc_sys
+            /*
+            SQLGetInfoW(SQL_DRIVER_ODBC_VER)
+            SQLGetInfoW(SQL_DRIVER_NAME)
+            */
+
+            let mut query: Vec<u16> = "select * from example".encode_utf16().collect();
+            query.push(0);
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLExecDirectW(stmt as HStmt, query.as_ptr(), SQL_NTS as i32)
+            );
+
+            // SQLGetFunctions is not available through odbc_sys
+            /*
+            SQLGetFunctions(SQL_API_SQLFETCHSCROLL)
+            */
+
+            let str_len_ptr = &mut 0;
+            const BUFFER_LENGTH: SmallInt = 300;
+            let output_buffer = &mut [0u16; (BUFFER_LENGTH as usize - 1)] as *mut _;
+
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLGetInfoW(
+                    conn_handle,
+                    InfoType::GetDataExtensions,
+                    output_buffer as Pointer,
+                    BUFFER_LENGTH,
+                    str_len_ptr
+                )
+            );
+
+            let column_count_ptr = &mut 0;
+            assert_eq!(
+                SqlReturn::SUCCESS,
+                SQLNumResultCols(stmt as HStmt, column_count_ptr)
+            );
+            let numeric_attribute_ptr = &mut 0;
+            const FIELD_TYPES: [Desc; 3] = [
```

Comment:
Can you rename this FIELD_IDENTIFIER or FIELD_ID? This way there is no confusion with SQL or Mongo types.